telemetry/prometheus/src/main/scala/io/buoyant/telemetry/prometheus/PrometheusTelemeter.scala

```diff
@@ -0,0 +1,124 @@
+ * histogram summaries directly off of the MetricsTree and assumes that stats
+ * are being snapshotted at some appropriate interval.
+ */
+class PrometheusTelemeter(metrics: MetricsTree) extends Telemeter with Admin.WithHandlers {
+
+  private[prometheus] val handler = Service.mk { request: Request =>
+    val response = request.response
+    response.mediaType = MediaType.Txt
+    val sb = new StringBuilder()
+    writeMetrics(metrics, sb)
+    response.contentString = sb.toString
+    Future.value(response)
+  }
+
+  val adminHandlers: Seq[Admin.Handler] = Seq(
+    Admin.Handler("/admin/metrics/prometheus", handler)
+  )
+
+  val stats = NullStatsReceiver
+  def tracer = NullTracer
+  def run(): Closable with Awaitable[Unit] = Telemeter.nopRun
+
+  private[this] val disallowedChars = "[^a-zA-Z0-9:]".r
+  private[this] def escapeKey(key: String) = disallowedChars.replaceAllIn(key, "_")
+
+  private[this] def formatLabels(labels: Seq[(String, String)]): String =
+    if (labels.nonEmpty) {
+      labels.map {
+        case (k, v) =>
+          s"""$k="$v""""
+      }.mkString("{", ", ", "}")
+    } else {
+      ""
+    }
+
+  private[this] def labelExists(labels: Seq[(String, String)], name: String) =
+    labels.exists(_._1 == name)
+
+  private[this] def writeMetrics(
+    tree: MetricsTree,
+    sb: StringBuilder,
+    prefix0: Seq[String] = Nil,
+    labels0: Seq[(String, String)] = Nil
+  ): Unit = {
+
+    // Re-write elements out of the prefix into labels
+    val (prefix1, labels1) = prefix0 match {
+      case Seq("rt", router) if !labelExists(labels0, "rt") =>
+        (Seq("rt"), labels0 :+ ("rt" -> router))
+      case Seq("rt", "dst", "path", path) if !labelExists(labels0, "dst_path") =>
+        (Seq("rt", "dst_path"), labels0 :+ ("dst_path" -> path))
+      case Seq("rt", "dst", "id", id) if !labelExists(labels0, "dst_id") =>
+        (Seq("rt", "dst_id"), labels0 :+ ("dst_id" -> id))
+      case Seq("rt", "dst_id", "path", path) if !labelExists(labels0, "dst_path") =>
+        (Seq("rt", "dst_id", "dst_path"), labels0 :+ ("dst_path" -> path))
+      case Seq("rt", "srv", srv) if !labelExists(labels0, "srv") =>
+        (Seq("rt", "srv"), labels0 :+ ("srv" -> srv))
+      case _ => (prefix0, labels0)
+    }
+
+    val key = escapeKey(prefix1.mkString(":"))
+
+    tree.metric match {
+      case c: Metric.Counter =>
+        sb.append(key)
+        sb.append(formatLabels(labels1))
+        sb.append(" ")
+        sb.append(c.get)
+        sb.append("\n")
+      case g: Metric.Gauge =>
+        sb.append(key)
+        sb.append(formatLabels(labels1))
+        sb.append(" ")
+        sb.append(g.get)
+        sb.append("\n")
+      case s: Metric.Stat =>
+        val summary = s.summary
+        for (
+          (stat, value) <- Seq(
+            "count" -> summary.count, "min" -> summary.min,
+            "max" -> summary.max, "sum" -> summary.sum, "avg" -> summary.avg
```

Comment:
looking at the `go_gc_duration_seconds` summary, maybe we should set min as `quantile="0"` and max as `quantile="1"`?

```
# HELP go_gc_duration_seconds A summary of the GC invocation durations.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 6.409200000000001e-05
go_gc_duration_seconds{quantile="0.25"} 0.00010941100000000001
go_gc_duration_seconds{quantile="0.5"} 0.000138934
go_gc_duration_seconds{quantile="0.75"} 0.000173474
go_gc_duration_seconds{quantile="1"} 0.00030891400000000003
go_gc_duration_seconds_sum 0.29322348800000003
go_gc_duration_seconds_count 1854
```